src/localized-text/LocalizedAppearanceEditor.component.js

```diff
@@ -35,42 +74,80 @@ class LocalizedTextEditor extends React.Component {
         this.state = {
             locale: settingsStore.state.keyUiLocale,
             localeName: LocalizedTextEditor.getLocaleName(settingsStore.state.keyUiLocale),
+            loading: true,
+            error: false,
         };
 
+        this.settings = null;
         this.handleChange = this.handleChange.bind(this);
         this.saveSettingsKey = this.saveSettingsKey.bind(this);
         this.getTranslation = context.d2.i18n.getTranslation.bind(context.d2.i18n);
     }
 
+    componentDidMount() {
+        this.getAppearanceSettings();
+    }
+
+    getAppearanceSettings(code) {
+        const locale = code || this.state.locale
+        const promise = locale === SYSTEM_DEFAULT
+            ? readDefaultAppearanceSettingsFromStore()
+            : this.fetchLocalizedAppearanceSettings(locale);
+        
+        promise.then(values => {
+            this.settings = LOCALIZED_SETTING_KEYS.reduce((acc, key, i) => {
```

Comment:
Why not put settings in state? Makes sense to me, since it's used in rendering.

Reply:
Very valid question. I tried that but some obscure ESLint rule in this project prevented me from doing so. I believe there is a rule that states: "All class methods need at least one reference to `this`. I could attach the LOCALIZED_SETTING_KEYS to the instance/class as well, that would solve the problem, but I actually thought that was more messy....

I think changing the ESLint rules is out-of-scope for this PR...

Reply:
But if you put settings in state you'd still be using `this` right?

```javascript
const settings = LOCALIZED_SETTING_KEYS.reduce((acc, key, i) => {
    acc[key] = values[i];
    return acc;
}, {});

this.setState({ settings, loading: false, error: false });
```

Reply:
Sorry, You were replying to a comment I removed, because I realised it didn't make much sense. I've now refactored the code to put settings in state and also to get rid of the loading state property